File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        indexMap = {}
        for i in range(len(nums)):
            if (target - nums[i]) not in indexMap:
                indexMap[nums[i]] = i
            else:
                return [indexMap[target - nums[i]], i]
        
        return [-1, -1]

File: 0001-two-sum/0001-two-sum.java
import java.util.HashMap;


class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> maps = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i ++) {
            if (maps.containsKey(target - nums[i])) {
                return new int[] {maps.get(target - nums[i]), i};
            } 
            maps.put(nums[i], i);
        }

        return new int[] {};
    }
}